pkg/controller/acmechallenges/update_test.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2022 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package acmechallenges
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clienttesting "k8s.io/client-go/testing"
+	featuretesting "k8s.io/component-base/featuregate/testing"
+
+	"github.com/cert-manager/cert-manager/internal/controller/feature"
+	"github.com/cert-manager/cert-manager/pkg/client/clientset/versioned/fake"
+	utilfeature "github.com/cert-manager/cert-manager/pkg/util/feature"
+	"github.com/cert-manager/cert-manager/test/unit/gen"
+)
+
+func TestUpdateObjectStandard(t *testing.T) {
+	runUpdateObjectTests(t)
+}
+
+func TestUpdateObjectSSA(t *testing.T) {
+	t.Skip("Server Side Apply cannot be tested because PatchType is not supported by the fake versioned client")
+	defer featuretesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, feature.ServerSideApply, true)()
+	runUpdateObjectTests(t)
```

Comment:
There's no support for Patch in the fake client. So I've left these tests here, but skipped.

Reply:
What do you mean by this? I see that there is a Patch for the fake challenge [here](https://github.com/cert-manager/cert-manager/blob/master/pkg/client/clientset/versioned/typed/acme/v1/fake/fake_challenge.go#L133-L142) and there is also a `Patch` action for test builder [here](https://github.com/kubernetes/client-go/blob/master/testing/actions.go#L167-L177)

Reply:
When I run the tests I get the following error:
```
$ go test ./pkg/controller/acmechallenges/... -run TestUpdateObjectSSA
...
--- FAIL: TestUpdateObjectSSA (0.00s)
    --- FAIL: TestUpdateObjectSSA/success (0.00s)
        update_test.go:154: Executing the function
        update_test.go:157: 
            	Error Trace:	update_test.go:157
            	Error:      	Received unexpected error:
            	            	[when updating the status: PatchType is not supported, when updating the finalizers: PatchType is not supported]

```

There are a couple of existing bug reports in client-go:

 * https://github.com/kubernetes/client-go/issues/970
 * https://github.com/kubernetes/client-go/issues/992

I'll add those links to the comment above the skipped test